### PR TITLE
[C-1548] Integrate native WalletConnect

### PR DESCRIPTION
**Pull request by raymondjacobson**
### Description

Vendors in wallet connect

TODO:
1. Move token dashboard sagas to common and wire up
Basically this code, but inside that saga
```
const connector = useWalletConnect()
connector.signMessage(...), etc.
```

2. Wire up phantom stuff (probably differently than is currently done b/c we never want to touch the privkey) into the same modal experience

3. Wire up solana phone as another option (but quite similar to phantom)

### Dragons

*Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?*

### How Has This Been Tested?

*Please describe the tests that you ran to verify your changes. Provide repro instructions & any configuration.*

Many tests necessary

### How will this change be monitored?

*For features that are critical or could fail silently please describe the monitoring/alerting being added.*

### Feature Flags ###

*Are all new features properly feature flagged? Describe added feature flags.*

### CODE REVIEW

**Comment by raymondjacobson** [comment]
Not ready to merge, but tagging you for vis @dylanjeffers , we can sync tm on this!

File: packages/mobile/package.json [hunk 1]
```diff
@@ -172,6 +178,7 @@
     "jest": "24.9.0",
     "jetifier": "1.6.8",
     "metro-react-native-babel-preset": "0.67.0",
+    "npm-force-resolutions": "0.0.10",
```
Comment (dylanjeffers) [inline packages/mobile/package.json:181]: cool!

File: packages/mobile/src/App.tsx [hunk 2]
```diff
@@ -79,21 +81,30 @@ const App = () => {
...
-                <Airplay />
-                <RootScreen isReadyToSetupBackend={isReadyToSetupBackend} />
-                <Drawers />
-                <Modals />
-                <Audio />
-                <OAuth />
-                <NotificationReminder />
-                <OfflineDownloader />
-              </NavigationContainer>
+              <WalletConnectProvider
```
Comment (dylanjeffers) [inline packages/mobile/src/App.tsx:84]: so do we need this wrapped in the root vs just around the wallet-connect screen?

File: packages/mobile/src/components/drawer/Drawer.tsx [hunk 3]
```diff
@@ -217,16 +219,17 @@ export const springToValue = ({
...
+      friction = 10 + 2 * (1 - Math.min(drawerHeight / FULL_DRAWER_HEIGHT, 1))
       break
   }
   Animated.spring(animation, {
     toValue: value,
     tension,
     friction,
     useNativeDriver: true,
-    velocity
+    velocity,
```
Comment (dylanjeffers) [inline packages/mobile/src/components/drawer/Drawer.tsx:231]: maybe we add a "animationConfig" prop for various things, but i think this is all good :)

File: packages/mobile/src/screens/wallet-connect/components/LinkedWallets.tsx [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+  )
+}
+
+export const LinkedWallets = () => {
+  const styles = useStyles()
+
+  const { connectedEthWallets, connectedSolWallets } =
+    useSelector(getAssociatedWallets)
+
+  const wallets = [
```
Comment (dylanjeffers) [inline packages/mobile/src/screens/wallet-connect/components/LinkedWallets.tsx:90]: will memoize this in future, maybe even we do this all in the selector

**Review by dylanjeffers — APPROVED** [review]
awesome lift here, main callout is if we need wallet-connect-provider at root of app, or whether we can isolate to wallet-connect-modal. in the future, that could help us from having to load extra js on initial load

File: packages/mobile/src/App.tsx (hunk 2 above)
Comment (raymondjacobson) [reply packages/mobile/src/App.tsx:84]: we don't! we can wrap further down!

Comment (dylanjeffers) [reply packages/mobile/src/App.tsx:84]: word word

File: packages/mobile/src/screens/wallet-connect/components/LinkedWallets.tsx (hunk 4 above)
Comment (raymondjacobson) [reply packages/mobile/src/screens/wallet-connect/components/LinkedWallets.tsx:90]: yeah sounds good. i originally was just iterating over and rendering, but then thought we should use a FlatList here for scrolling and it expects a single array of data

**Pull request merged**